### PR TITLE
PMM-9224: Allow connect Mongo arbiter to PMM when it requires authentication

### PR DESCRIPTION
**Pull request by dliakhov**
PMM-9224

Enable connection to arbiter when there authentication is enabled.

And change `listDatabase` command to `getDiagnosticData` to check whether mongodb client is authenticated or not, because on arbiter instance  `listDatabase` doesn't work (this mongo instance can't see databases in the replica set).

Link to the Feature Build: https://github.com/Percona-Lab/pmm-submodules/pull/3260

- https://github.com/percona/mongodb_exporter/pull/655

### CODE REVIEW

File: agent/utils/mongo_fix/mongo_fix.go [hunk 1]
```diff
@@ -40,6 +40,10 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) {
 	if username != "" || password != "" {
 		clientOptions.Auth.Username = username
 		clientOptions.Auth.Password = password
+
+		// set this flag to connect to arbiter when there authentication is enabled
+		b := true
+		clientOptions.AuthenticateToAnything = &b //nolint:staticcheck
```
Comment (BupycHuk) [inline agent/utils/mongo_fix/mongo_fix.go:46]: we have `pointer` library for this kind of stuff

Comment (dliakhov) [reply agent/utils/mongo_fix/mongo_fix.go]: thanks, fixed

**Review by BupycHuk — COMMENTED** [review]
does getDiagnosticData work on any instance?

File: agent/utils/mongo_fix/mongo_fix.go [hunk 2]
```diff
@@ -16,6 +16,7 @@
 package mongo_fix
 
 import (
+	"github.com/AlekSi/pointer"
```
[the remark this answers is not in the document]
Comment (dliakhov) [reply agent/utils/mongo_fix/mongo_fix.go]: Fixed.

**Comment by dliakhov** [comment]
> does getDiagnosticData work on any instance?

I've tested it on the primary, secondary and arbiter instances - it works fine.

Also this command works in the mongodb_exporter as one of the sources to fetch info about mongodb, so it should work on every type of the instance.

**Review by BupycHuk — APPROVED** [review]

**Review by 0leksii — APPROVED** [review]

**Pull request merged**